tests/gradients/core/test_pulse_gradient.py

```diff
@@ -245,26 +211,42 @@ def test_warning_no_trainable_params(self, num_meas):
 
     def test_raises_non_pulse_marked_as_trainable(self):
         """Test that an empty gradient is returned when there are no trainable parameters."""
-        import jax.numpy as jnp
-
-        ops = [qml.RX(jnp.array(0.4), wires=0)]
+        ops = [qml.RX(0.4, wires=0)]
         tape = qml.tape.QuantumScript(ops, measurements=[qml.expval(qml.PauliZ(0))])
         tape.trainable_params = [0]
         with pytest.raises(ValueError, match="stoch_pulse_grad does not support differentiating"):
             stoch_pulse_grad(tape)
 
     def test_raises_for_non_pauli_term(self):
-        """Test that an error is raised if a ParametrizedHamiltonian contains a paramatrized
+        """Test that an error is raised if a ParametrizedEvolution contains a paramatrized
         term that is not a Pauli word."""
-        import jax.numpy as jnp
-
         ham = qml.dot([qml.pulse.constant], [qml.PauliX(0) + qml.PauliY(2)])
-        ops = [qml.evolve(ham)([jnp.array(0.152)], 0.3)]
+        ops = [qml.evolve(ham)([0.152], 0.3)]
         tape = qml.tape.QuantumScript(ops, measurements=[qml.expval(qml.PauliZ(0))])
         tape.trainable_params = [0]
         with pytest.raises(ValueError, match="stoch_pulse_grad currently only supports Pauli"):
             stoch_pulse_grad(tape)
 
+    def test_raises_use_broadcasting_with_broadcasted_tape(self):
+        """Test that an error is raised if the option `use_broadcasting` is activated
+        for a tape that already is broadcasted."""
+        ham = qml.dot([qml.pulse.constant], [qml.PauliX(0)])
+        ops = [qml.evolve(ham, return_intermediate=True)([0.152], 0.3)]
+        tape = qml.tape.QuantumScript(ops, measurements=[qml.expval(qml.PauliZ(0))])
+        tape.trainable_params = [0]
+        with pytest.raises(ValueError, match="Broadcasting is not supported for tapes that"):
+            stoch_pulse_grad(tape, use_broadcasting=True)
+
+    def test_raises_pulse_with_multiple_time_steps(self):
+        """Test that an error is raised if a differentiated pulse already has multiple
+        time steps given."""
+        ham = qml.dot([qml.pulse.constant], [qml.PauliX(0)])
+        ops = [qml.evolve(ham, return_intermediate=True)([0.152], 0.3)]
+        tape = qml.tape.QuantumScript(ops, measurements=[qml.expval(qml.PauliZ(0))])
+        tape.trainable_params = [0]
+        with pytest.raises(ValueError, match="Broadcasting is not supported for tapes that"):
+            stoch_pulse_grad(tape, use_broadcasting=True)
+
```

Comment:
These tests are identical, but the docstring of the 2nd test implies that the 2nd test should be doing something different.

Reply:
Oops, we decided to actually allow this feature, so this test should simply be deleted.